common/lib/xmodule/xmodule/modulestore/django.py

```diff
@@ -243,10 +245,46 @@ class ModuleI18nService(object):
     i18n service.
 
     """
+    def __init__(self, block=None):
+        """
+        We are using this block for things like locating the block's translation domain (PO+MO files). Here we attempts
+         to look up the provided string in the XBlock's own domain translation catalog, currently expected to be:
+            <xblock_root>/conf/locale/<language>/LC_MESSAGES/<domain>.po
+        If ModuleI18nService can't locate the domain translation catalog then we fall-back onto
+        django.utils.translation.ugettext, which will attempt to find a matching string in the
+        LMS' own domain translation catalog -- effectively achieving translation by coincidence.
+        """
+        self.translator = django.utils.translation
+        if block:
+            xblock_resource = block.__class__.unmixed_class.__module__
+            xblock_locale_dir = '/conf/locale'
+            xblock_locale_path = resource_filename(xblock_resource, xblock_locale_dir)
+            xblock_domain = 'django'
+            selected_language = get_language()
+            try:
+                self.translator = gettext.translation(
+                    xblock_domain,
+                    xblock_locale_path,
+                    [to_locale(selected_language if selected_language else settings.LANGUAGE_CODE)]
+                )
+            except IOError:
+                pass
 
```

Comment:
We should add a comment describing the exceptional scenario and why it's acceptable to pass -- could be a simple as something like "Fall back to the default Django translator if the XBlock translator is not found."